modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/ServerWorker.java

```diff
@@ -464,7 +464,19 @@ public MessageContext createMessageContext(MessageContext msgContext, SourceRequ
 //        msgContext.setProperty(Constants.OUT_TRANSPORT_INFO, this);
         
         NHttpServerConnection conn = request.getConnection();
-        
+
+
+        //observability measurements
+        boolean correlationEnabled = conn.getContext().getAttribute(PassThroughConstants.CORRELATION_LOG_STATE_PROPERTY).toString().equals(PassThroughConstants.CORRELATION_ENABLE_STATE);
+        if(correlationEnabled) {
+            msgContext.setProperty(PassThroughConstants.CORRELATION_ID, conn.getContext().getAttribute(PassThroughConstants.CORRELATION_ID));
+            msgContext.setProperty(PassThroughConstants.CORRELATION_TIME, conn.getContext().getAttribute(PassThroughConstants.CORRELATION_TIME));
+        }
+        msgContext.setProperty(PassThroughConstants.CORRELATION_LOG_STATE_PROPERTY, conn.getContext().getAttribute(PassThroughConstants.CORRELATION_LOG_STATE_PROPERTY));
+        //observability code ends here
+
```

Comment:
Fix formatting issues.
Can we make the check for correlationEnabled a one time process ?
Shall we use correlationLoggingEnabled instead of correlationEnabled